README.md

```diff
@@ -1,1218 +1,533 @@
-  configure script tries to figure this out by compiling and running a few
-  small test programs. This is of course not possible when cross-compiling.
-  You can use the `--with-fp-layout` option to tell the configure script which
-  conversion method to assume. Valid arguments are:
-
-    * `nothing`    (12345678 -> 12345678)
-    * `endianflip` (12345678 -> 87654321)
-    * `intswap`    (12345678 -> 56781234)
-
-
-Contact
--------
-
-  Please use GitHub to report bugs and submit pull requests:
-  <https://github.com/collectd/collectd/>.
-  See CONTRIBUTING.md for details.
-
-  For questions, development information and basically all other concerns please
-  send an email to collectd's mailing list at
-  (list at collectd.org).
-
-  For live discussion and more personal contact visit us in IRC, we're in
-  channel #collectd on freenode.
-
-
-Author
-------
-
-  Florian octo Forster (octo at collectd.org),
-  Sebastian tokkee Harl (sh at tokkee.org),
-  and many other [authors](AUTHORS) and [contributors](https://github.com/collectd/collectd/graphs/contributors).
+- POSIX threads (pthread) library
+
+### Building from Source
+
+```bash
+git clone https://github.com/collectd/collectd.git
+cd collectd
+./build.sh
+./configure
+make
+sudo make install
+```
+
+### Package Managers
+
+```bash
+# Debian/Ubuntu
+apt-get install collectd
+
+# RHEL/CentOS
+dnf install collectd
+
+# macOS (Homebrew)
+brew install collectd
+```
+
+## Configuration
+
+```mermaid
+graph LR
+    A[Configuration File] -->|Read| B[Parser]
+    B -->|Configure| C[Global Settings]
+    B -->|Load| D[Plugins]
+    D -->|Configure| E[Plugin Settings]
+    E -->|Apply| F[Runtime Configuration]
+```
+
+### Basic Configuration Example
+
+```apache
+# Global settings
+Hostname "myhost"
+Interval 10
+
+# Load plugins
+LoadPlugin cpu
+LoadPlugin memory
+
+# Plugin configuration
+<Plugin disk>
+  Disk "/dev/sda"
+  IgnoreSelected false
+</Plugin>
+```
+
+### Important Settings
+
+- `Hostname`: Sets the hostname for collected data
+- `Interval`: Default interval between data collections (in seconds)
+- `ReadThreads`: Number of threads used for data collection
+- `WriteThreads`: Number of threads used for data dispatch
+
+## Usage
+
+### Starting collectd
+
+```bash
+# Start as daemon
+collectd
+
+# Run in foreground
+collectd -f
+
+# Test configuration
+collectd -t
+
+# Use alternate config file
+collectd -C /path/to/config
+```
+
+### Signals
+
+- `SIGINT`, `SIGTERM`: Graceful shutdown
+- `SIGHUP`: Restart daemon
+- `SIGUSR1`: Flush all plugins
+
+## Plugin System
+
+```mermaid
+graph TD
+    A[Input Plugins] -->|Collect| B[Core]
+    B -->|Process| C[Filter Plugins]
+    C -->|Transform| D[Output Plugins]
+    
+    subgraph Input Plugins
+    E[CPU]
+    F[Memory]
+    G[Disk]
+    H[Network]
+    end
+    
+    subgraph Output Plugins
+    I[RRDTool]
+    J[CSV]
+    K[Network]
+    L[Databases]
+    end
+```
+
```

Comment:
IMHO replicates overall arch diagram too much.   I would suggest moving parts of this under relevant subsections:
* Input plugin subgraph -> collectd Core
* collecd Core -> Output plugins subgraph

That way it's more consistent with the networking section.